test/docker-e2e/e2e_upgrade_test.go

```diff
@@ -256,3 +304,192 @@ func getSignalQueryClient(node tastoratypes.ChainNode) (signaltypes.QueryClient,
 	}
 	return nil, nil, fmt.Errorf("GRPC connection is nil")
 }
+
+// getICAHostQueryClient returns an icahosttypes.QueryClient for the provided node.
+// If the node is a docker ChainNode with a live *grpc.ClientConn, it is reused.
+// Returns an error if no gRPC connection is available.
+func getICAHostQueryClient(node tastoratypes.ChainNode) (icahosttypes.QueryClient, func(), error) {
```

Comment:
[blocking] what is the second return argument here for? In the happy path it's an empty function, in the failure path it's nil. It seems completely useless.